src/components/ModelSteps/Scan.jsx

```diff
@@ -1,23 +1,86 @@
-import React, { useState } from 'react'
+import React, { useState, useRef, useCallback, useEffect } from 'react'
 
+import { useClient, models } from 'cozy-client'
+import ActionMenu from 'cozy-ui/transpiled/react/ActionMenu'
 import DialogActions from 'cozy-ui/transpiled/react/DialogActions'
 import { ButtonLink } from 'cozy-ui/transpiled/react/Button'
 import Camera from 'cozy-ui/transpiled/react/Icons/Camera'
 import { useI18n } from 'cozy-ui/transpiled/react/I18n'
 import FileInput from 'cozy-ui/transpiled/react/FileInput'
+import List from 'cozy-ui/transpiled/react/MuiCozyTheme/List'
+import ListItem from 'cozy-ui/transpiled/react/MuiCozyTheme/ListItem'
+import ListItemIcon from 'cozy-ui/transpiled/react/MuiCozyTheme/ListItemIcon'
+import Typography from 'cozy-ui/transpiled/react/Typography'
+import Icon from 'cozy-ui/transpiled/react/Icon'
+import FolderMoveto from 'cozy-ui/transpiled/react/Icons/FolderMoveto'
+import FilePicker from 'cozy-ui/transpiled/react/FilePicker'
+import Alerter from 'cozy-ui/transpiled/react/Alerter'
 
 import { useStepperDialog } from 'src/components/Hooks/useStepperDialog'
 import CompositeHeader from 'src/components/CompositeHeader/CompositeHeader'
 import AcquisitionResult from 'src/components/ModelSteps/AcquisitionResult'
 import IlluGenericNewPage from 'src/assets/icons/IlluGenericNewPage.svg'
+import PhoneUpload from 'src/assets/icons/PhoneUpload.svg'
+
+const { fetchBlobFileById } = models.file
+
+// TODO Waiting for this type of filter to be implemented on the FilePicker side
+// https://github.com/cozy/cozy-ui/issues/2026
+const validFileType = file => {
+  const regexValidation = /(image\/*)|(application\/pdf)/
+  return regexValidation.test(file.type)
+}
 
 const Scan = ({ currentStep }) => {
   const { t } = useI18n()
+  const client = useClient()
   const { illustration, text } = currentStep
   const [file, setFile] = useState(null)
+  const actionBtnRef = useRef()
   const { alreadyScan } = useStepperDialog()
 
-  const onFileChange = file => file && setFile(file)
+  const [isFilePickerOpen, setIsFilePickerOpen] = useState(false)
+  const [fileId, setFileId] = useState('')
+  const [isPickFileModalOpen, setIsPickFileModalOpen] = useState(false)
+
+  const closeFilePicker = useCallback(() => setIsFilePickerOpen(false), [])
+
+  const openPickerFileModal = useCallback(() => {
+    setIsFilePickerOpen(true)
+  }, [])
+  const closePickFileModal = useCallback(
+    () => setIsPickFileModalOpen(false),
+    []
+  )
+
+  const onPickFile = useCallback(() => {
```

Comment:
nit: I expected this method to be called when the user picks a file, not when they click on "I already have a picture". Maybe this method name can be improved?